Cross-compiling `ring v0.17.8` dependency fails

`verus-analyzer` (on the upcoming `infra` branch) uses `verusfmt` as a library.  When the (very complex) GitHub action runs to generate a release of `verus-analyzer`, one of the commands executed is:
```
$ cargo build --manifest-path ./crates/rust-analyzer/Cargo.toml --bin rust-analyzer --target aarch64-unknown-linux-gnu --release
```
This eventually fails with:
```
   Compiling ring v0.17.8
   Compiling ra-ap-rustc_index_macros v0.53.0
   Compiling futures-core v0.3.30
   Compiling profile v0.0.0 (/home/runner/work/verus-analyzer/verus-analyzer/crates/profile)
   Compiling chalk-derive v0.97.0
   Compiling ra-ap-rustc_index v0.53.0
The following warnings were emitted during compilation:

warning: ring@0.17.8: In file included from crypto/curve25519/curve25519.c:22:
warning: ring@0.17.8: In file included from include/ring-core/mem.h:60:
warning: ring@0.17.8: In file included from include/ring-core/base.h:64:
warning: ring@0.17.8: In file included from /usr/lib/llvm-11/lib/clang/11.0.0/include/stdint.h:52:
warning: ring@0.17.8: /usr/include/stdint.h:26:10: fatal error: 'bits/libc-header-start.h' file not found
warning: ring@0.17.8: #include <bits/libc-header-start.h>
warning: ring@0.17.8:          ^~~~~~~~~~~~~~~~~~~~~~~~~~
warning: ring@0.17.8: 1 error generated.

error: failed to run custom build command for `ring v0.17.8`
```
Running `cargo tree` seems to point the finger at `verusfmt`, in so far as it's the only package that uses `ring`.  Here's a simplified version of the relevant output (with arrows added for emphasis):
```
│   └── verusfmt v0.4.0
│       ├── axoupdater v0.6.9
│       │   ├── axoasset v1.0.0
│       │   │   ├── reqwest v0.12.5
│       │   │   │   ├── hyper-rustls v0.27.2
│       │   │   │   │   ├── futures-util v0.3.30 (*)
│       │   │   │   │   ├── http v1.1.0 (*)
│       │   │   │   │   ├── rustls v0.23.12
│       │   │   │   │   │   ├── once_cell v1.19.0
│       │   │   │   │   │   ├── ring v0.17.8                             <----
│       │   │   │   │   │   │   ├── cfg-if v1.0.0
│       │   │   │   │   │   │   ├── getrandom v0.2.15 (*)
│       │   │   │   │   │   │   ├── spin v0.9.8
│       │   │   │   │   │   │   └── untrusted v0.9.0
│       │   │   │   │   │   │   [build-dependencies]
│       │   │   │   │   │   │   └── cc v1.0.97
│       │   │   │   │   │   ├── rustls-pki-types v1.7.0
│       │   │   │   │   │   ├── rustls-webpki v0.102.6
│       │   │   │   │   │   │   ├── ring v0.17.8 (*)                     <----
│       │   │   │   │   │   │   ├── rustls-pki-types v1.7.0
│       │   │   │   │   │   │   └── untrusted v0.9.0
│       │   │   │   │   │   ├── subtle v2.6.1
│       │   │   │   │   │   └── zeroize v1.8.1
```
Can we address this by bumping a version on one of these packages?  Or by [updating `Cargo.toml`](https://stackoverflow.com/questions/35711044/how-can-i-specify-binary-only-dependencies), so that the library version doesn't depend on `axoupdater` (which seems unnecessary for the library)?

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -19,7 +19,7 @@ exclude = [
 ]
 
 [dependencies]
-axoupdater = { version = "0.7.2", features = ["blocking", "github_releases"], default-features = false }
+axoupdater = { version = "0.7.2", features = ["blocking", "github_releases"], default-features = false, optional = true }
 clap = { version = "4.3.11", features = ["derive"] }
 fs-err = "3.0.0"
 miette = { version = "7.2.0", features = ["fancy"] }
@@ -38,6 +38,13 @@ insta = { version = "1.30.0" }
 similar = { version = "2.2.1" }
 stacker = "0.1.15"
 glob-macro = { path = "glob-macro" }
+
+[features]
+# We enable the "optional" dependencies, that are only necessary for the binary,
+# as default-enabled features. If verusfmt is being used as a library, then that
+# program can turn default features off in order to obtain a more stripped-down
+# version of the library.
+default = ["axoupdater"]
 
 # Spend more time on initial compilation in exchange for faster runs
 [profile.dev.package.insta]

